docs/csharp/tour-of-csharp/classes-and-objects.md

```diff
@@ -0,0 +1,728 @@
+* Operators
+    - Conversions and expression operators supported by the class
+* Constructors
+    - Actions required to initialize instances of the class or the class itself
+* Finalizers
+    - Actions to perform before instances of the class are permanently discarded
+* Types
+    - Nested types declared by the class
+
+## Accessibility
+
+Each member of a class has an associated accessibility, which controls the regions of program text that are able to access the member. There are five possible forms of accessibility. These are summarized below.
+
+* `public`
+	- Access not limited
+* `protected`
+	- Access limited to this class or classes derived from this class
+* `internal`
+	- Access limited to this program
+* `protected internal`
+	- Access limited to this program or classes derived from this class
+* `private`
+	- Access limited to this class
+
+## Type parameters
+
+A class definition may specify a set of type parameters by following the class name with angle brackets enclosing a list of type parameter names. The type parameters can then be used in the body of the class declarations to define the members of the class. In the following example, the type parameters of `Pair` are `TFirst` and `TSecond`:
+
+```csharp
+public class Pair<TFirst,TSecond>
+{
+	public TFirst First;
+	public TSecond Second;
+}
+```
+
+A class type that is declared to take type parameters is called a ***generic class type***. Struct, interface and delegate types can also be generic.
+When the generic class is used, type arguments must be provided for each of the type parameters:
+
+```csharp
+Pair<int,string> pair = new Pair<int,string> { First = 1, Second = “two” };
+int i = pair.First;     // TFirst is int
+string s = pair.Second; // TSecond is string
+```
+
+A generic type with type arguments provided, like `Pair<int,string>` above, is called a ***constructed type***.
+
+## Base classes
+
+A class declaration may specify a base class by following the class name and type parameters with a colon and the name of the base class. Omitting a base class specification is the same as deriving from type object. In the following example, the base class of `Point3D` is `Point`, and the base class of `Point` is `object`:
+
+```csharp
+public class Point
+{
+	public int x, y;
+	public Point(int x, int y) 
+	{
+		this.x = x;
+		this.y = y;
+	}
+}
+public class Point3D: Point
+{
+	public int z;
+	public Point3D(int x, int y, int z) : 
+		base(x, y) 
+	{
+		this.z = z;
+	}
+}
+```
+
+A class inherits the members of its base class. Inheritance means that a class implicitly contains all members of its base class, except for the instance and static constructors, and the finalizers of the base class. A derived class can add new members to those it inherits, but it cannot remove the definition of an inherited member. In the previous example, `Point3D` inherits the `x` and `y` fields from `Point`, and every `Point3D` instance contains three fields, `x`, `y`, and `z`.
```

Comment:
This being the overview, I want to keep it as simple as possible. I'm OK with omissions, as long as the topic is correct.